02-cpu-configure/Makefile

```
@@ -0,0 +1,16 @@
+CFLAGS = -Wall -Werror -O2
+PREFIX ?= /usr/local
+
+all: pcie-remove pcie-rescan
+
+pcie-remove: pcie-remove.c
+	cc $(CFLAGS) $< -o $@
+
+pcie-rescan: pcie-rescan.c
+	cc $(CFLAGS) $< -o $@
```

```suggestion
CFLAGS = -Wall -Werror -O2
PREFIX ?= /usr/local

all: pcie-remove pcie-rescan
```

You can omit the whole commands for compiling these files, the automatic rules take care of it, all you need is `all: pcie-rescan pcie-remove`. This allows anyone who's using this makefile to override all compilation options without you having to remember to include them in your commands, i.e. `CC`, `CPPFLAGS`, `LDFLAGS`, `LDLIBS`.

---

Makefile magic. Is this compatible with other `make` utilities other than the GNU's `make`?

---

Yes, the rules are defined by [POSIX](https://pubs.opengroup.org/onlinepubs/9799919799.2024edition/utilities/make.html#tag_20_76_13_09)

GNU make adds `LDLIBS` and `CPPFLAGS`, at least. It seems `bmake` supports some of the additions as well.

---

nice